src/diffusers/pipeline_utils.py

```diff
@@ -709,5 +710,83 @@ def progress_bar(self, iterable):
+        >>> pipe.set_scheduler("euler_discrete")
+        ```
+        """
+        schedulers = {k: type(v) for k, v in self.components.items() if isinstance(v, SchedulerMixin)}
+
+        if isinstance(scheduler_type, str) and len(set(schedulers.values())) > 1:
+            raise ValueError(
+                f"The pipeline {self} contains the schedulers {schedulers}. Please make sure to provide a dictionary"
+                f" that maps the componet names {schedulers.keys()} to scheduler types. Providing just one scheduler"
+                f" type {scheduler_type} is ambiguous."
+            )
+        elif isinstance(scheduler_type, dict):
+            is_type_scheduler = {k: k in schedulers for k in scheduler_type.keys()}
+            if not all(is_type_scheduler.values()):
+                raise ValueError(
+                    "The following component names are not schedulers"
+                    f" {[k for k, v in is_type_scheduler.items() if v == False]}. Please make sure to only set new"
+                    f" scheduler types for {schedulers.keys()}."
+                )
+
+        scheduler_mapping = (
+            scheduler_type if isinstance(scheduler_type, dict) else {next(iter(schedulers.keys())): scheduler_type}
+        )
+
+        for component_name, scheduler_type in scheduler_mapping.items():
+            if isinstance(scheduler_type, SchedulerType):
+                scheduler_type = scheduler_type.name
+
+            scheduler_class = SCHEDULER_TYPE_TO_CLASS_MAPPING.get(scheduler_type, None)
+            current_scheduler = getattr(self, component_name)
+
+            if scheduler_class is None:
+                raise ValueError(
+                    f"{scheduler_type} does not exist, make sure to chose a scheduler type from"
+                    f" {', '.join(SCHEDULER_TYPE_TO_CLASS_MAPPING.keys())}."
+                )
+
+            if scheduler_class not in current_scheduler._compatible_classes and scheduler_class != type(
+                current_scheduler
+            ):
+                diffusers_library = importlib.import_module(__name__.split(".")[0])
+                _compatible_class_types = [
+                    CLASS_TO_SCHEDULER_TYPE_MAPPING[getattr(diffusers_library, c)]
+                    for c in current_scheduler._compatible_classes
+                ]
+                logger.warn(
+                    f"Changing scheduler from type {CLASS_TO_SCHEDULER_TYPE_MAPPING[type(current_scheduler)]} to an"
+                    f" uncompatible scheduler type {scheduler_type}. This is very likely going to lead to incorrect"
+                    f" predictions when running the pipeline. Make sure to set {component_name} to a scheduler of type"
+                    f" {[' ,'.join(_compatible_class_types)]}."
+                )
+
+            scheduler_config = current_scheduler.config
+            scheduler_init_dict, _ = scheduler_class.extract_init_dict(scheduler_config)
```

Comment:
This extracts the required config of the new scheduler from the old config @keturn